src/main/java/com/limechain/sync/fullsync/FullSyncMachine.java

```diff
@@ -154,7 +186,11 @@ private void executeBlocks(List<Block> receivedBlockDatas, BlockTrieAccessor blo
         for (Block block : receivedBlockDatas) {
             log.fine("Block number to be executed is " + block.getHeader().getBlockNumber());
 
-            blockState.addBlock(block);
+            try {
+                blockState.addBlock(block);
+            } catch (BlockNodeNotFoundException ignored) {
```

Comment:
BlockState addBlock calls BlockTree adblock, which checks for a parent node, but because we warped, we don't have the current parent and that's why it will throw this exception.